app/test/vesting_test.go

```diff
@@ -0,0 +1,550 @@
+				expectedSpendableBalAfterTx -= testfactory.BaseAccountDefaultBalance
+			}
+
+			assert.EqualValues(s.T(),
+				expectedSpendableBalAfterTx,
+				balancesAfter.AmountOf(app.BondDenom).Int64(),
+				"spendable balance must be equal")
+
+			/*--------*/
+
+			// test delegation of only the locked account(s)
+			if !mustSucceed {
+				s.testDelegation(cctx, tt.accName)
+				s.testClaimDelegationReward(cctx, tt.accName)
+			}
+		})
+	}
+}
+
+func (s *VestingModuleTestSuite) TestGenesisPeriodicVestingAccounts() {
+	// initial unlocked allocation to pay the gas fees
+	initBalanceForGasFee := sdk.NewCoin(app.BondDenom, sdk.NewInt(10))
+
+	startTime := tmtime.Now()
+	periods := vestingtypes.Periods{
+		vestingtypes.Period{Length: int64(10), Amount: sdk.Coins{sdk.NewInt64Coin(app.BondDenom, 2500)}},
+		vestingtypes.Period{Length: int64(8), Amount: sdk.Coins{sdk.NewInt64Coin(app.BondDenom, 2500)}},
+		vestingtypes.Period{Length: int64(8), Amount: sdk.Coins{sdk.NewInt64Coin(app.BondDenom, 2500)}},
+		vestingtypes.Period{Length: int64(10), Amount: sdk.Coins{sdk.NewInt64Coin(app.BondDenom, 2500)}},
+	}
+
+	const accName = "period_vesting_0"
+	kr := testfactory.GenerateKeyring()
+	bacc, coins := testfactory.NewBaseAccount(kr, accName)
+	pva := vestingtypes.NewPeriodicVestingAccount(bacc, coins, startTime.Unix(), periods)
+
+	gsOpt := func(gs map[string]json.RawMessage) map[string]json.RawMessage {
+		vAccs := authtypes.GenesisAccounts{pva}
+		vBals := []banktypes.Balance{
+			{
+				Address: pva.GetAddress().String(),
+				Coins:   pva.GetOriginalVesting().Add(initBalanceForGasFee),
+			},
+		}
+
+		var err error
+		gs, err = AddAccountsToGenesisState(gs, vAccs...)
+		assert.NoError(s.T(), err)
+
+		gs, err = AddBalancesToGenesisState(gs, vBals...)
+		assert.NoError(s.T(), err)
+
+		return gs
+	}
+
+	cctx, _, _ := s.NewNetworkWithGenesisOpt(kr, gsOpt)
+
+	assert.NoError(s.T(), cctx.WaitForNextBlock())
+
+	accAddress := getAddress(accName, cctx.Keyring).String()
+
+	// Test account details correctness
+	cli := authtypes.NewQueryClient(cctx.GRPCClient)
+	res, err := cli.Account(context.Background(), &authtypes.QueryAccountRequest{
+		Address: accAddress,
+	})
+	assert.NoError(s.T(), err)
+
+	var qAcc vestingtypes.PeriodicVestingAccount
+	err = qAcc.Unmarshal(res.Account.Value)
+	assert.NoError(s.T(), err)
+
+	// Checking the queried account data
+	assert.Equal(s.T(), accAddress, qAcc.GetAddress().String(), "account address must match")
+	assert.EqualValues(s.T(), startTime.Unix(), qAcc.GetStartTime(), "start time must match")
+	assert.EqualValues(s.T(),
+		testfactory.BaseAccountDefaultBalance,
+		qAcc.GetOriginalVesting().AmountOf(app.BondDenom).Int64(),
+		"original vesting must match")
+	assert.EqualValues(s.T(), periods, qAcc.VestingPeriods, "periods must match")
+
+	/*--------*/
+
+	// Test available balance
+	for i := 0; i < 2; i++ { // We just let one period to be unlocked to keep some locked amount to test other stuff
```

Comment:
Since I wanted to use one periodic vesting account to test multiple things, there is a confusion. Let me separate tests as you suggested to make thing clear